### ISSUE TITLE
SKPath.Bounds returns the wrong result

### ISSUE THREAD

**Issue by gentledepp** [mono/SkiaSharp#150]
Hi!

I recently came across an issue of SKPath.GetBounds(out SKRect)

The following SVG was created in InkScape.
If you open it with that tool and select the single path that is containted, you'll see, that its bounds are actually exactly aligned with the x-axis (so x = 0)

However, SKPath.GetBounds() returns a "Left" value of "-16"

```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!-- Created with Inkscape (http://www.inkscape.org/) -->

<svg
   xmlns:dc="http://purl.org/dc/elements/1.1/"
   xmlns:cc="http://creativecommons.org/ns#"
   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
   xmlns:svg="http://www.w3.org/2000/svg"
   xmlns="http://www.w3.org/2000/svg"
   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
   width="300mm"
   height="300mm"
   viewBox="0 0 1062.9921 1062.9921"
   id="svg2"
   version="1.1"
   inkscape:version="0.91 r13725"
   sodipodi:docname="Bends.svg">
  <title
     id="title4138">Vorlage</title>
  <defs
     id="defs4">
    <clipPath
       clipPathUnits="userSpaceOnUse"
       id="clipPath7866">
      <path
         d="m 54,769.8 486.96,0 0,-677.4 -486.96,0 0,677.4 z"
         id="path7868"
         inkscape:connector-curvature="0"
         style="clip-rule:evenodd" />
    </clipPath>
    <clipPath
       clipPathUnits="userSpaceOnUse"
       id="clipPath7856">
      <path
         d="m 54,769.8 0,0 486.96,0 0,-677.4 -486.96,0 0,677.4 z"
         id="path7858"
         inkscape:connector-curvature="0"
         style="clip-rule:evenodd" />
    </clipPath>
    <clipPath
       clipPathUnits="userSpaceOnUse"
       id="clipPath7816">
      <path
         d="m 54,769.8 486.96,0 0,-677.4 -486.96,0 0,677.4 z"
         id="path7818"
         inkscape:connector-curvature="0"
         style="clip-rule:evenodd" />
    </clipPath>
    <clipPath
       clipPathUnits="userSpaceOnUse"
       id="clipPath7826">
      <path
         d="m 540.96,769.8 0,-677.4 -486.96,0 0,677.4 486.96,0 z"
         id="path7828"
         inkscape:connector-curvature="0"
         style="clip-rule:evenodd" />
    </clipPath>
  </defs>
  <sodipodi:namedview
     id="base"
     pagecolor="#ffffff"
     bordercolor="#666666"
     borderopacity="1.0"
     inkscape:pageopacity="0.0"
     inkscape:pageshadow="2"
     inkscape:zoom="5.6568544"
     inkscape:cx="29.882178"
     inkscape:cy="1028.4464"
     inkscape:document-units="px"
     inkscape:current-layer="bends_solid_001"
     showgrid="true"
     objecttolerance="10"
     gridtolerance="10"
     inkscape:window-width="1536"
     inkscape:window-height="801"
     inkscape:window-x="1912"
     inkscape:window-y="-8"
     inkscape:window-maximized="1"
     inkscape:snap-smooth-nodes="true"
     inkscape:object-nodes="true"
     inkscape:snap-others="true"
     inkscape:snap-bbox="false">
    <inkscape:grid
       type="axonomgrid"
       id="grid4136"
       units="px"
       empspacing="5"
       spacingy="20.000001" />
  </sodipodi:namedview>
  <metadata
     id="metadata7">
    <rdf:RDF>
      <cc:Work
         rdf:about="">
        <dc:format>image/svg+xml</dc:format>
        <dc:type
           rdf:resource="http://purl.org/dc/dcmitype/StillImage" />
        <dc:title>Vorlage</dc:title>
      </cc:Work>
    </rdf:RDF>
  </metadata>
  <g
     id="bends_solid_001"
     transform="translate(17.320509,30.000003)">
    <path
       style="fill:none;fill-rule:evenodd;stroke:#000000;stroke-width:2;stroke-linecap:butt;stroke-linejoin:miter;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1"
       d="m -6.2157825e-7,-25.814698 c -34.64102137842175,19.9999998 0,40 0,40 l 0,0 0,0"
       id="path4304"
       inkscape:connector-curvature="0"
       sodipodi:nodetypes="cccc" />
  </g>
</svg>
```

Seems that there is some bug in the SKPath binding? (is that possible?)

**Comment by gentledepp** [comment]
I dug a little deeper into this and it seems, that it takes the first control point of a cubic bezier curve for calculating the bounding box, although that point is not visible!

**Comment by mattleibow** [comment]
@gentledepp It seems Skia does this: https://fiddle.skia.org/c/8972161759375a6a1463ebaa0b80cad4

Can you confirm that this is what is happening. If so, maybe check on the skia forums, they may be better suited to answer your questions: https://groups.google.com/forum/#!forum/skia-discuss

**Comment by gentledepp** [comment]
Dude! That "skia.fiddle" is awesome!
I just woke up this morning and realized that I actually sent you SVG.... stupid me. I am sorry for that.
That exactly is the bug we stumbled across.

**Comment by gentledepp** [comment]
I filed a bug here:
https://bugs.chromium.org/p/skia/issues/detail?id=5751

**Comment by gentledepp** [comment]
I got a response to the bug I filed.
Seems that it is no bug, but the intended behavior.

As I, however, need the exact "visible" bounding box, there is another API method available:

SKPathOps "TightBounds" which does what we need.
see: /** Set the resulting rectangle to the tight bounds of the path.
    @param path The path measured.
    @param result The tight bounds of the path.
    @return True if the bounds could be computed.
  */
bool SK_API TightBounds(const SkPath& path, SkRect* result); 

However, it seems that this API is not bound yet
https://github.com/google/skia/blob/master/include/pathops/SkPathOps.h

I will open a new issue for that :)
